frontend/src/pages/ProgressionChecker/ProgressionChecker.jsx

```diff
@@ -42,29 +43,34 @@ const ProgressionChecker = () => {
   const [isLoading, setIsLoading] = useState(true);
 
   const {
-    programCode, specialisation, minor,
+    programCode, majors, minor,
   } = useSelector((state) => state.degree);
 
   const [structure, setStructure] = useState({});
 
   useEffect(() => {
     // get structure of degree
     const fetchStructure = async () => {
-      try {
-        const res = await axios.get(`/programs/getStructure/${programCode}/${specialisation}${minor && `/${minor}`}`);
-        setStructure(res.data.structure);
-      } catch (err) {
-        // eslint-disable-next-line no-console
-        console.log(err);
-      }
+      await Promise.all(majors.map(async (major) => {
```

Comment:
Think we don't need to do a map or multiple promises? Iirc the backend supports double degree with this: 
```
const res = await axios.get(`/programs/getStructure/${programCode}/${majors.join("+")}${minor && `/${minor}`}`);
```

Reply:
o my, dam that makes it a lot easier then - thanks :) will just revert to the previous structure logic